src/xmlstr2xmlnode.js

```diff
@@ -4,8 +4,9 @@ const util = require('./util');
 const buildOptions = require('./util').buildOptions;
 const xmlNode = require('./xmlNode');
 const TagType = {OPENING: 1, CLOSING: 2, SELF: 3, CDATA: 4};
-let regx =
-  '<((!\\[CDATA\\[([\\s\\S]*?)(]]>))|(([\\w:\\-._]*:)?([\\w:\\-._]+))([^>]*)>|((\\/)(([\\w:\\-._]*:)?([\\w:\\-._]+))\\s*>))([^<]*)';
+const regx =
+  '<((!\\[CDATA\\[([\\s\\S]*?)(]]>))|((NAME:)?(NAME))([^>]*)>|((\\/)(NAME)\\s*>))([^<]*)'
```

Comment:
Can you please recheck this regex?

Expected regex is : `<((!\\[CDATA\\[([\\s\\S]*?)(]]>))|((NAME:)?(NAME))([^>]*)>|((\\/)((NAME:)?(NAME))\\s*>))([^<]*)`

Reply:
This is redundant. The previous one can also be removed. Colon is a valid character everywhere in the name, so there's no reason to have it explicitly stated in the regex, especially when it is not mandatory [i.e. `(NAME:)?`].

Reply:
`:` can not be repeated in name. Hence, taking in NAME regex but not in 2nd regex will not be correct. However, as the validator is separate functionality we can have it in the way you did it. But in this case, I see, there is no point of having in first part as well. right?

`'<((!\\[CDATA\\[([\\s\\S]*?)(]]>))|(NAME)([^>]*)>|((\\/)(NAME)\\s*>))([^<]*)'`

Considering this, we can even move this logic to the validator `const nameRegexp = '[' + nameStartChar + '][' + nameChar + ']*'`, instead of keeping it in parser.

Reply:
From the spec:

> Note:
> 
> The Namespaces in XML Recommendation [XML Names] assigns a meaning to names containing colon characters. Therefore, authors should not use the colon in XML names except for namespace purposes, but **XML processors must accept the colon as a name character**.
> 
> ...
> 
> Document authors are encouraged to use names which are meaningful words or combinations of words in natural languages, and to avoid symbolic or white space characters in names. **Note that COLON, HYPHEN-MINUS, FULL STOP (period), LOW LINE (underscore), and MIDDLE DOT are explicitly permitted**.

Regarding the first appearance in the regex, like I said, it is needed for the ignoreNamespace option.

Reply:
Agree.. I believe this needs to be handled properly by comparing the namespace name instead of its syntax only (probably in version FXP v4).